cmd/print.go

```diff
@@ -55,7 +55,11 @@ var printCmd = &cobra.Command{
 		if err != nil {
 			return err
 		}
-		namespaceFilter, err := getNamespaceFilter(namespace, allNamespaces)
+		inputFileMode := false
+		if inputFile != "" {
+			inputFileMode = true
+		}
+		namespaceFilter, err := getNamespaceFilter(namespace, allNamespaces, inputFileMode)
 		if err != nil {
 			return err
 		}
```

Comment:
I think it would be cleaner if you returned a predefined error at getNamespaceFilter (something like ErrNoContext), and then checked if that specific error was returned.
If such an error occurs, and we are reading from files return an empty string, otherwise return an error to the user.

Just my opinion.

Reply:
+1